src/utils/queries.js

```diff
@@ -1,16 +1,24 @@
 import { Q } from 'cozy-client'
 
 import { FILES_DOCTYPE } from 'src/doctypes'
+import papersJSON from 'src/constants/papersDefinitions.json'
+
+const papersLabel = papersJSON.papersDefinitions.map(paper => paper.label)
 
 export const getAllPapers = {
   definition: () =>
-    Q(FILES_DOCTYPE).where({
-      type: 'file',
-      trashed: false,
-      'metadata.qualificationsAttributes': {
-        $exists: true
-      }
-    }),
+    Q(FILES_DOCTYPE)
+      .where({
+        type: 'file',
+        trashed: false,
+        'metadata.qualification': {
+          $exists: true,
+          label: {
+            $in: papersLabel
+          }
+        }
+      })
+      .sortBy([{ name: 'asc' }]),
```

Comment:
can you add an `indexFields` please? https://docs.cozy.io/en/tutorials/data/queries/#index-fields from the more restrictive fields to the more opened

Reply:
Ans since you'll index by metadata.qualification there is no need to $exists:true since if it doesn't exists, it'll not be indexed

Reply:
It's a good practice to explicitly index fields with `.indexFields` to avoid relying on the attribute order in the `where` clause (cozy-client will automatically create an index based on this order). As @Crash-- implied, the order matters ;)
Furthermore, it's more efficient to use [partial index](https://docs.cozy.io/en/tutorials/data/advanced/#partial-indexes) for "static" predicates, such as `trashed: false`.
So I would suggest something like this:
```
      .where({
        'metadata.qualification': {
          label: {
            $in: papersLabel
          }
        }
      })
      .partialIndex({
          type: 'file',
          trashed: false
      })
      .indexFields(['name', 'metadata.qualification'])
      .sortBy([{ name: 'asc' }])
  ```

Reply:
Thanks for your explanations/links 👍